### PR TITLE
float literals now parse using musl's 128 bit float code

### PR DESCRIPTION
**Pull request by andrewrk**
Fixes float literals not having 128 bit precision. Related: #2093.

How I did this is I grabbed musl's strtold code, and manually edited the code to assume all the `#defines` where `LDBL_MANT_DIG == 113`. Then I replaced `long double` with `float128_t` (from softfloat) and painstakingly converted all the float operations to use softfloat lib calls.

I also had to provide f128 versions of `scalbn` and `fmodf` using similar strategies. Also musl uses its own internal `FILE` struct, so I just gave it what it wanted and made sure it wouldn't conflict with stdio.h.

I believe this also fixes some incorrect range checks on float literals. Previously we had 
`const a = 0x1.0p16384;` as an example of too big and  `const a = 0x1.0p-16384;` as an example of too small. But the code ported from musl has those in bounds and I had to make different values which returned `ERANGE`.

I hope this compiles with MSVC :crossed_fingers:

### CODE REVIEW

File: src/tokenizer.cpp [hunk 1]
```diff
@@ -293,10 +293,10 @@ static void cancel_token(Tokenize *t) {
...
 static void end_float_token(Tokenize *t) {
-    if (t->radix == 10) {
+    if (t->radix == 10 || t->radix == 16) {
         uint8_t *ptr_buf = (uint8_t*)buf_ptr(t->buf) + t->cur_tok->start_pos;
         size_t buf_len = t->cur_tok->end_pos - t->cur_tok->start_pos;
-        if (bigfloat_init_buf_base10(&t->cur_tok->data.float_lit.bigfloat, ptr_buf, buf_len)) {
+        if (bigfloat_init_buf(&t->cur_tok->data.float_lit.bigfloat, ptr_buf, buf_len)) {
             t->cur_tok->data.float_lit.overflow = true;
         }
         return;
```
Comment (tiehuis) [inline src/tokenizer.cpp]: We can remove the proceeding following float parsing code now, right? Only radix 10 and 16 is valid for float literals.

Comment (tiehuis) [reply src/tokenizer.cpp]: Ahh my mistake, just noticed we accept octal and binary float literals. Interesting, didn't know that.

Comment (andrewrk) [reply src/tokenizer.cpp]: I did that at first but it broke tests and so I put it back and filed #2093

File: src/parse_f128.c [hunk 2]
```diff
@@ -0,0 +1,1039 @@
...
+        ui64_to_f128M(1000000000, &const_f128);
+        float128_t mul_y;
+        f128M_mul(&const_f128, &y, &mul_y);
+        float128_t x_f128;
+        ui32_to_f128M(x[a+i & MASK], &x_f128);
+        f128M_add(&mul_y, &x_f128, &y);
+	}
+
+	//y *= sign;
+    mul_eq_f128_int(&y, sign);
```
Comment (daurnimator) [inline src/parse_f128.c]: weird indentation

File: src/parse_f128.c [hunk 3]
```diff
@@ -0,0 +1,1039 @@
...
+            float128_t mod_result = fmodf128(frac, one);
+            if (f128M_eq(&mod_result, &zero)) {
+                //frac++;
+                add_eq_f128_dbl(&frac, 1.0);
+            }
+        }
+	}
+
+	//y += frac;
+    {
```
Comment (daurnimator) [inline src/parse_f128.c]: more weird indentation here and following.

File: src/parse_f128.c [hunk 4]
```diff
@@ -0,0 +1,1039 @@
...
+		if (e2+LDBL_MANT_DIG>emax || (denormal && !f128M_eq(&frac, &zero)))
+			errno = ERANGE;
+	}
+
+	return scalbnf128(y, e2);
+}
+
+static float128_t hexfloat(struct MuslFILE *f, int bits, int emin, int sign, int pok)
+{
+    float128_t zero;
```
Comment (daurnimator) [inline src/parse_f128.c]: indentation.....

**Pull request merged**